packages/linea-ens-resolver/test/testL1ResolverLocal.spec.ts

```diff
@@ -0,0 +1,670 @@
+ * to aggregation-proofs-limit=15
+ * - Then run make fresh-start-all
+ * - In the linea-ens monorepo: Run cd packages/linea-ens-resolver/
+ * - Then run pnpm i
+ * - pnpm test:local
+ */
+describe("Crosschain Resolver Local", () => {
+  let l1Provider: JsonRpcProvider;
+  let l2Provider: JsonRpcProvider;
+  let verifier: Contract;
+  let target: Contract;
+  let l2Resolver: Contract;
+  let wrapper: Contract;
+  let baseRegistrar: Contract;
+  let rollup: Contract;
+  let l2factoryContract: Contract;
+  let signerL1,
+    signerL2,
+    signerL1Address,
+    signerL2Address,
+    l2ResolverAddress,
+    wrapperAddress;
+  let lastSetupTxBlockNumber = BigInt(0);
+  let sendTransactionsPromise: NodeJS.Timeout;
+
+  before(async () => {
+    // Setup providers and signers
+    l1Provider = new ethers.JsonRpcProvider(L1_NODE_URL, L1_CHAIN_ID, {
+      staticNetwork: true,
+    });
+    l2Provider = new ethers.JsonRpcProvider(L2_NODE_URL, L2_CHAIN_ID, {
+      staticNetwork: true,
+    });
+    signerL1 = new ethers.Wallet(SIGNER_L1_PK, l1Provider);
+    signerL1Address = await signerL1.getAddress();
+    signerL2 = new ethers.Wallet(SIGNER_L2_PK, l2Provider);
+    signerL2Address = await signerL2.getAddress();
+    rollup = await ethers.getContractAt(
+      "RollupMock",
+      ROLLUP_CONTRACT_ADDR,
+      signerL1
+    );
+
+    // Setup CCIP Gateway
+    const shomeiNode = new ethers.JsonRpcProvider(
+      SHOMEI_NODE_URL,
+      L2_CHAIN_ID,
+      {
+        staticNetwork: true,
+      }
+    );
+    const gateway = makeL2Gateway(
+      l1Provider as unknown as JsonRpcProvider,
+      l2Provider,
+      await rollup.getAddress(),
+      shomeiNode
+    );
+    const server = new Server();
+    gateway.add(server);
+    const app = server.makeApp("/");
+    const getUrl = FetchRequest.createGetUrlFunc();
+    ethers.FetchRequest.registerGetUrl(async (req: FetchRequest) => {
+      if (req.url != "test:") return getUrl(req);
+
+      const r = request(app).post("/");
+      if (req.hasBody()) {
+        r.set("Content-Type", "application/json").send(
+          ethers.toUtf8String(req.body)
+        );
+      }
+      const response = await r;
+      return {
+        statusCode: response.statusCode,
+        statusMessage: response.ok ? "OK" : response.statusCode.toString(),
+        body: ethers.toUtf8Bytes(JSON.stringify(response.body)),
+        headers: {
+          "Content-Type": "application/json",
+        },
+      };
+    });
+
+    // Deploy and configure contracts
+    const ens = await deployContract("ENSRegistry", signerL1);
+    baseRegistrar = await deployContract(
+      "BaseRegistrarImplementation",
+      signerL1,
+      await ens.getAddress(),
+      ethers.namehash("eth")
+    );
+    const baseRegistrarAddress = await baseRegistrar.getAddress();
+    await baseRegistrar.addController(signerL1Address).then((tx) => tx.wait());
```

Comment:
across the files there is a mix of syntax  which should be standardised
```
await tx.wait(); 
```
vs. 
```
.then((tx) => tx.wait())
```

Reply:
This is a small simplification to have a one liner instead of:
```
let tx = await baseRegistrar.addController(signerL1Address);
await tx.await();
```

Reply:
I just made the change